lnwallet/wallet.go

```diff
@@ -2546,7 +2561,18 @@ func (c *CoinSource) ListCoins(minConfs int32,
 	}
 
 	var coins []wallet.Coin
+
 	for _, utxo := range utxos {
+		// If there is a filter function supplied all utxos not adhering
+		// to these conditions will be discared.
+		if c.allowUtxo != nil && !c.allowUtxo(*utxo) {
```

Comment:
Yes agree we might consider doing this. I liked doing it way higher up the design because I realized I would need to funnel the argument into every function call, so because I started with the single funding flow I was happy to introduce the change at a higher level. Only later I found out that the psbt part can only be done by funneling the argument  all the way down to the btcwallet lib. Maybe you are right having the filtering all in one place might be more cleaner. I open to rework it. Keep in mind that the filter logic will live in two different functions tho in the btcwallet lib. (`findEligibleOutputs` and `ListUnspent`). The latter is also exported so it might cause a bigger API change. Wdyt ?

Reply:
The psbt case had already options in the call `CreateSimpleTx` so it was easy to introduce.